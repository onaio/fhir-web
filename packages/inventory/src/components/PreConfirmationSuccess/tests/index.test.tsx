import { mount, shallow } from 'enzyme';
import React from 'react';
import { PreConfirmationSuccess } from '..';

describe('pre confirmation success', () => {
  it('renders without crashing', () => {
    shallow(<PreConfirmationSuccess />);
  });

  it('works correctly', () => {
    const cancelMock = jest.fn();
    const commitMock = jest.fn(() => Promise.resolve({}));
    const props = {
      onCancel: cancelMock,
      onCommitInventory: commitMock,
      rowsProcessed: 0,
      filename: 'uploaded.csv',
    };

    const wrapper = mount(<PreConfirmationSuccess {...props} />);

    // full card snapshot
    expect(wrapper.text()).toMatchSnapshot('full card snapshot');

    // simulate onCommit callback
    wrapper.find('button#confirm-commit').simulate('click');

    wrapper.update();

    expect(commitMock).toHaveBeenCalled();

    expect(cancelMock).not.toHaveBeenCalled();

    // simulate cancel

    wrapper.find('button#cancel-commit').simulate('click');

    wrapper.update();

    expect(cancelMock).toHaveBeenCalled();
  });
});
