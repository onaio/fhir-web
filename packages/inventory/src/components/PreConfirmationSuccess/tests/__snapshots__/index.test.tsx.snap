// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pre confirmation success works correctly: full card snapshot 1`] = `"“uploaded.csv” ready0 inventory items will be added to service points. Do you wish to proced?Proceed with adding inventoryCancel"`;
